app/src/main/java/ch/epfl/sdp/blindly/main_screen/MatchPageFragment.kt

```diff
@@ -34,7 +40,37 @@ class MatchPageFragment : Fragment() {
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
         // Inflate the layout for this fragment
-        return inflater.inflate(R.layout.fragment_match_page, container, false)
+        val view = inflater.inflate(R.layout.fragment_match_page, container, false)
+
+        val intent = Intent(context, MatchActivity::class.java)
+        val profiles = createProfiles()
+        intent.putParcelableArrayListExtra("EXTRA_MATCH_PROFILES", profiles as ArrayList<out Parcelable>)
+        val match_activity_button = view.findViewById<Button>(R.id.button_to_match)
+        match_activity_button.setOnClickListener{startActivity(intent)}
+
+        return view
+    }
+
+    /**
+     * create profiles for demo purpose
+     * this function have to be replaced with calls to the matching algorithms and retrieve porfiles
```

Comment:
Pretty useless comment since this will disappear but there's a typo in profiles :p

Reply:
just to be sure to remove it after we bind it with the database

Reply:
I didn't mean your comment was useless, I was saying my comment about the typo was useless because you're going to remove it later xD the second word profiles is written porfiles